src/main.cpp

```diff
@@ -33,7 +33,7 @@ int main(void) {
   auto screen = Screen::Create(Dimension::Full(), Dimension::Fit(document));
   Render(screen, document);
 
-  std::cout << screen.ToString() << '\0' << std::endl;
+  std::cout << screen.ToString() << "\0\n";
```

Comment:
I worry developers will start not using `std::endl` and not understand the standard output needs to be flushed at some point for the content to be displayed on the terminal.

Could you please revert this line?

Reply:
I have reverted the line back to the original.

---

As an inexperienced C++ developer, I would appreciate an explanation of why `std::endl` is required here.

From what I understand, `std::cout` is flushed automatically when the buffer is full, whereas `std::endl` explicitly flushes the buffer, which incurs a performance penalty. Therefore, `std::endl` should be used only when the output needs to be displayed immediately (e.g., for debugging or before asking for input from the user).

Given this, I typically avoid using `std::endl` and use `\n` instead. For example, in my code analyzer program, I am not interested in receiving the output immediately; performance is more important.

Should I explicitly flush the output in `main()` before the program ends? For example:

```cpp
int main()
{
    try {
        for (const auto &file : files) {
            // Load file from disk, parse it, and return the report as a long string
            std::cout << get_report(file) << '\n';
        }
    }
    catch (const std::exception &e) {
        std::cerr << std::string(e.what()) + '\n';  // std::cerr, no need to flush
        return EXIT_FAILURE;
    }
    catch (...) {
        std::cerr << "Unknown error\n";  // std::cerr again, no need to flush
        return EXIT_FAILURE;
    }
    std::cout << std::flush;  // Explicitly flush the buffer
    return 0;
}
```

Reply:
When the program ends, the buffer is flushed. So that is not necessary.

I was worried about user copy-pasting the line without "std::endl" and not understanding why there terminal wasn't displaying anything on:
```cpp
While(true) {
  ...
  std::cout << screen.ToString() << "\0\n";
  WaitForEvents()
}
```

Reply:
Thank you for your assistance. I found the information online to be quite contradictory.

If you have the time, could you please review my usage of FTXUI? I aim to include it in the list of projects using your library, but I'm unsure if my implementation is correct. Specifically, I wanted to display a loading screen while the JSON file is loading, so I created a separate thread. I am new to multithreading, so there might be some issues.

You can find the relevant source code [here](https://github.com/ryouze/shogun/blob/main/src/app.cpp#L159). Everything is contained within the `app::run()` function for simplicity.

If you don't have the time, no worries, I understand. Your help with `std::endl` was already greatly appreciated!